tests/pyfunc/test_pyfunc_model_with_type_hints.py

```diff
@@ -452,7 +452,8 @@ def python_model(x) -> list[str]:
         model_info = mlflow.pyfunc.log_model(
             "model", python_model=python_model, input_example=["a"]
         )
-    assert model_info.signature is None
+    assert model_info.signature.inputs == Schema([ColSpec(type=DataType.string)])
+    assert model_info.signature.outputs == Schema([ColSpec(type=DataType.string, name=0)])
```

Comment:
we may want to change the behavior here since the output signature is ignored.

Reply:
Yes, while I think we should warn users when they only have output type hints

Reply:
Let's follow up on this.